src/services/continuedev/core/llm/llms/OpenRouter.ts

```diff
@@ -0,0 +1,143 @@
+import { ChatCompletionCreateParams } from "openai/resources/index"
+import { LLMOptions } from "../../index.js"
+import { osModelsEditPrompt } from "../templates/edit.js"
+import { OpenAI } from "./OpenAI"
+
+class OpenRouter extends OpenAI {
+	static override providerName = "openrouter"
+	protected supportsReasoningField = true
+	protected supportsReasoningDetailsField = true
+	static defaultOptions: Partial<LLMOptions> = {
+		apiBase: "https://openrouter.ai/api/v1/",
+		model: "gpt-4o-mini",
```

Comment:
dont we want to keep using codestral?

Reply:
We are using Codestral as default, this file comes from the ContinueDev repo that have gpt-4o-mini as default